### PR TITLE
Chore: Remove unused Loki and Cloudwatch syntax providers 

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
Following up on https://github.com/grafana/grafana/pull/29381#issuecomment-736739839, this PR removes syntax providers in Loki & Cloudwatch as we are passing syntax directly to prism. 

In Loki, we are introducing new variable - labelsLoaded that is used instead of syntaxLoaded to set the cascader's button text.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/LogsQueryEditor.tsx [hunk 1]
```diff
@@ -36,11 +34,6 @@ export const CloudWatchLogsQueryEditor = memo(function CloudWatchLogsQueryEditor
     };
   }
 
-  const { isSyntaxReady, syntax } = useCloudWatchSyntax(
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/cloudwatch/components/LogsQueryEditor.tsx:39]: We are passing syntax directly to PrismSlate plugin, so this is not needed anymore.

File: public/app/plugins/datasource/loki/components/AnnotationsQueryEditor.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const LokiAnnotationsQueryEditor = memo(function LokiAnnotationQueryEdito
     to: Date.now(),
   };
 
-  const { isSyntaxReady, setActiveOption, refreshLabels, syntax, logLabelOptions } = useLokiSyntaxAndLabels(
+  const { setActiveOption, refreshLabels, logLabelOptions, labelsLoaded } = useLokiLabels(
     datasource.languageProvider,
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/AnnotationsQueryEditor.tsx:26]: labelsLoaded is replacing the isSyntaxReady to determine the text of cascader button

File: public/app/plugins/datasource/loki/components/useLokiLabels.test.ts [hunk 3]
```diff
@@ -1,30 +1,90 @@
...
-    const { result, waitForNextUpdate } = renderHook(() => useLokiLabels(languageProvider, true, rangeMock));
+    const { result, waitForNextUpdate } = renderHook(() => getLokiLabels(languageProvider, true, rangeMock));
     expect(result.current.logLabelOptions).toEqual(['initial']);
     act(() => result.current.refreshLabels());
     await waitForNextUpdate();
     expect(result.current.logLabelOptions).toEqual(logLabelOptionsMock);
   });
 });
+
+describe('useLokiLabels hook', () => {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/useLokiLabels.test.ts:38]: Tests from the useLokiLabelsAndSyntax that test the log labels fetching and refreshing

File: public/app/plugins/datasource/loki/components/useLokiLabels.ts [hunk 4]
```diff
@@ -6,23 +6,50 @@ import { CascaderOption } from '@grafana/ui';
 import LokiLanguageProvider from 'app/plugins/datasource/loki/language_provider';
 import { useRefMounted } from 'app/core/hooks/useRefMounted';
 
+/**
+ * Initialise the language provider. Returns a languageProviderInitialized boolean cause there does not seem other way
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/useLokiLabels.ts:10]: Based on removed code from useLokiLabelsAndSyntax

**Review by aocenas — APPROVED** [review]
Nice :+1:

**Pull request merged**